### PR TITLE
feat(): add file-system provider

### PR DESCRIPTION
**Pull request by DanielaDurandBartolo**
# Summary

- [ ] Bug fix (corrects a fault without impacting current features)
- [X] New feature (adds a new functionality without affecting the existing ones)
- [ ] Major change (adds or modifies a feature that impacts existing functionalities)

- @mdf/file-system-provider: create provider for managing file system operations using fs from Node.js  .

## What has changed?

- @mdf/file-system-provider: create provider for managing file system operations using fs from Node.js  .

### Author's Checklist Before Publication

- [X] I have verified that the modification works correctly.
- [X] The modification adheres to the standards set by the Mytra Control DevOps group.
- [X] The modifications do not include keys, tokens, or confidential information in the pipeline.
- [X] Corresponding unit tests have been conducted to cover the new code and/or the tests that are no longer necessary have been removed.

### CODE REVIEW

File: packages/providers/file-system/src/provider/Port.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  /** Return the underlying port instance */
+  public get client(): Client {
+    return this.instance;
+  }
+  /** Return the port state as a boolean value, true if the port is available, false in otherwise */
+  public get state(): boolean {
+    return this.instance.isErrored;
+  }
+  /** Initialize the port instance */
+  public async start(): Promise<void> {
```
Comment (cjimenezsaiz) [inline packages/providers/file-system/src/provider/Port.ts:37]: - No use anonymous handlers for events.
- You should 'off' the event handler in stop

File: packages/providers/file-system/src/config/schema.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2024 Mytra Control S.L. All rights reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be found in the LICENSE file
+ * or at https://opensource.org/licenses/MIT.
+ */
+
+import Joi from 'joi';
+
+export const schema = Joi.object({
```
Comment (cjimenezsaiz) [inline packages/providers/file-system/src/config/schema.ts:10]: - this schema has more fields that the fields you need, check it using other providers

File: packages/providers/file-system/src/Client/FileSystemManager.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  /** Unique identifier */
+  private readonly uuid: string = v4();
+  /** Logger instance for deep debugging tasks */
+  private readonly logger: LoggerInstance;
+  /** Validation error, if exist */
+  private _error?: Multi;
+  /**
+   * Creates a new instance of the file manager
+   * @param name - Service name
+   * @param options - Service setup options
```
Comment (cjimenezsaiz) [inline packages/providers/file-system/src/Client/FileSystemManager.ts:50]: - you need to doc the logger

File: packages/providers/file-system/src/Client/FileSystemManager.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+  /** Reset error when last operation was successfull*/
+  private resetError(): void {
+    if (this._error) {
+      this._error = undefined;
+      this.emit('no-error');
+    }
+  }
+
+  /** Appends data to a file.
```
Comment (cjimenezsaiz) [inline packages/providers/file-system/src/Client/FileSystemManager.ts:92]: - fix the comment format

File: packages/providers/file-system/src/Client/FileSystemManager.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+import { v4 } from 'uuid';
+import {
+  CopyOptions,
+  FileSystemManagerOptions,
+  ReadDirOptions,
+  ReadOptions,
+  WriteOptions,
+} from './types';
+
+export declare interface FileSystemManager {
```
Comment (cjimenezsaiz) [inline packages/providers/file-system/src/Client/FileSystemManager.ts:22]: -  the client should manage the file, not give an interface to manage the file externaly, this means that the class should offer the public function 'append', the rest is part of the logic of this class. 
- this logic should rotate the file, move the file to other folder, move old files to other folder ...

**Comment by cjimenezsaiz** [comment]
- Change the provider nane to: jsonl-file-store
 - fix the rest of comments

**Review by cjimenezsaiz — CHANGES_REQUESTED** [review]

File: packages/providers/file-system/README.md
[the remark this answers is not in the document]
Comment (DanielaDurandBartolo) [reply packages/providers/file-system/README.md:1]: Updated. Added rotation options.

File: packages/providers/file-system/src/config/schema.ts (hunk 2 above)
Comment (DanielaDurandBartolo) [reply packages/providers/file-system/src/config/schema.ts]: Done.

File: packages/providers/file-system/src/Client/FileSystemManager.ts (hunk 3 above)
Comment (DanielaDurandBartolo) [reply packages/providers/file-system/src/Client/FileSystemManager.ts]: Done.

File: packages/providers/file-system/src/Client/FileSystemManager.ts (hunk 4 above)
Comment (DanielaDurandBartolo) [reply packages/providers/file-system/src/Client/FileSystemManager.ts]: Done.

File: packages/providers/file-system/src/Client/FileSystemManager.ts (hunk 5 above)
Comment (DanielaDurandBartolo) [reply packages/providers/file-system/src/Client/FileSystemManager.ts]: Done.

**Comment by cjimenezsaiz** [comment]
Pull request closed and moved to release branch

**Pull request closed without merging**